File: Java-Spring/src/main/java/org/codecop/lcdnumbers/Line.java
package org.codecop.lcdnumbers;

import java.util.Objects;

/**
 * Value Object of an LCD digit line.
 */
public final class Line {

    private final String line;

    public Line(String line) {
        this.line = line;
    }

    public Line scaleHorizontal(ScalingRepeater repeater, Scaling scaling) {
        char[] chars = line.toCharArray();
        return new Line(chars[0] + repeater.repeat(chars[1], scaling) + chars[2]);
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof Line)) {
            return false;
        }
        Line that = (Line) other;
        return this.line.equals(that.line);
    }

    @Override
    public int hashCode() {
        return line.hashCode();
    }

    @Override
    public String toString() {
        return line;
    }
}

File: Kotlin-Dagger2/src/main/kotlin/org/codecop/lcdnumbers/Line.kt
package org.codecop.lcdnumbers

/**
 * Value Object of an LCD digit line.
 */
data class Line(private val line: String) {
    fun scaleHorizontal(repeater: ScalingRepeater, scaling: Scaling): Line {
        val chars = line.toCharArray()
        return Line(chars[0] + repeater.repeatChar(chars[1], scaling) + chars[2])
    }
    
    override fun toString() = line
}